phoenicis-library/src/main/java/org/phoenicis/library/ShortcutRunner.java

```diff
@@ -42,11 +46,20 @@ public void run(String shortcutName, List<String> arguments, Consumer<Exception>
     public void run(ShortcutDTO shortcutDTO, List<String> arguments, Consumer<Exception> errorCallback) {
         final InteractiveScriptSession interactiveScriptSession = scriptInterpreter.createInteractiveSession();
 
+        ScriptEngineManager m = new ScriptEngineManager();
+        ScriptEngine engine = m.getEngineByName("graal.js");
+        Invocable inv = (Invocable) engine;
         interactiveScriptSession.eval("include([\"engines\", \"wine\", \"shortcuts\", \"reader\"]);",
                 ignored -> interactiveScriptSession.eval("new ShortcutReader()", output -> {
-                    final ScriptObjectMirror shortcutReader = (ScriptObjectMirror) output;
-                    shortcutReader.callMember("of", shortcutDTO);
-                    shortcutReader.callMember("run", arguments);
+                    final Object shortcutReader = (Object) output;
```

Comment:
My guess is, that when you're using `graal.js` you'll receive a `Value` object here.
Have you checked?
If this is the case you can use the more comfortable methods of the `Value` class, see: https://www.graalvm.org/sdk/javadoc/org/graalvm/polyglot/Value.html

Reply:
What issues did you have?
For the include:
Is it possible to replace `ScriptObjectMirror` with `Value`?
The `Value` class has a method `as(Class<T>)` which returns an object of type `T` can we use this?

Reply:
I have to try. Actually this is exactly why I want to have the branch on upstream: that you can try yourself.